File: backend/app/schemas/analytics.py
from pydantic import BaseModel
from typing import List, Optional, Dict, Any

class ActivationKPISchema(BaseModel):
    profile_completeness_pct: float
    dossier_completeness_pct: float
    match_explainability_pct: float
    verified_contacts_count: int

class GTMKPISchema(BaseModel):
    total_buyers_monitored: int
    grade_a_matches: int
    grade_b_matches: int
    active_signals_count: int
    total_customs_teu: float
    enterprise_mrr_pipeline_usd: float

class ExecutiveKPIDashboardResponse(BaseModel):
    timestamp: str
    active_exporter: str
    exporter_origin: str
    activation: ActivationKPISchema
    gtm: GTMKPISchema
    recent_agent_runs: int
    crm_exports_count: int

File: frontend/src/api/analytics.ts
import { fetchApi } from './client';

export interface ActivationKPIs {
  profile_completeness_pct: number;
  dossier_completeness_pct: number;
  match_explainability_pct: number;
  verified_contacts_count: number;
}

export interface GTMKPIs {
  total_buyers_monitored: number;
  grade_a_matches: number;
  grade_b_matches: number;
  active_signals_count: number;
  total_customs_teu: number;
  enterprise_mrr_pipeline_usd: number;
}

export interface ExecutiveKPIDashboardResponse {
  timestamp: string;
  active_exporter: string;
  exporter_origin: string;
  activation: ActivationKPIs;
  gtm: GTMKPIs;
  recent_agent_runs: number;
  crm_exports_count: number;
}

export const getExecutiveKPIsApi = () =>
  fetchApi<ExecutiveKPIDashboardResponse>('/api/v1/analytics/kpis');
